### PR TITLE
feat: new internal function

### PR DESCRIPTION
**Pull request by maggiben**
### What does this PR do?
Prevents from making duplicated comments

### What issues does this PR fix or reference?
@W-8752767@

### CODE REVIEW

File: packages/kit/src/nodash/internal.ts [hunk 1]
```diff
@@ -38,6 +38,24 @@ export function lowerFirst(value?: string): Optional<string> {
   return value && value.charAt(0).toLowerCase() + value.slice(1);
 }
 
+/**
+ * Formats a camel case style `string` into a title case.
+ *
+ * @param text Text to transform.
+ */
+export function camelCaseToTitleCase(str: string): string;
+export function camelCaseToTitleCase(str?: string): Optional<string>;
```
Comment (amphro) [inline packages/kit/src/nodash/internal.ts:47]: Why even allow str to be optional?

Comment (maggiben) [reply packages/kit/src/nodash/internal.ts:47]: Good I'll remove the option both as param and return values.

**Review by amphro — APPROVED** [review]

File: packages/kit/test/nodash/internal.test.ts [hunk 2]
```diff
@@ -91,4 +91,12 @@ describe('nodash internal', () => {
       expect(_.upperFirst('abc')).to.equal('Abc');
     });
   });
+
+  describe('format', () => {
+    it('returns a title case string given a camel case one', () => {
```
Comment (peternhale) [inline packages/kit/test/nodash/internal.test.ts:96]: @maggiben The input to this test already has first character as upper case. Could you add another test case that supplies a string with lowercase first character?

Also, what happens if the string is already formatted in title case?

**Review by peternhale — CHANGES_REQUESTED** [review]

File: packages/kit/test/nodash/internal.test.ts (hunk 2 above)
Comment (maggiben) [reply packages/kit/test/nodash/internal.test.ts:96]: @peternhale I added a few more tests the original function comes from another repo: https://github.com/salesforcecli/plugin-org/blob/main/src/shared/utils.ts

File: packages/kit/test/nodash/internal.test.ts [hunk 3]
```diff
@@ -91,4 +91,17 @@ describe('nodash internal', () => {
...
+      let result = _.camelCaseToTitleCase('FooBar');
+      expect(result).to.equal('Foo Bar');
+
+      result = _.camelCaseToTitleCase('fooBar');
+      expect(result).to.equal('Foo Bar');
+
+      result = _.camelCaseToTitleCase('Foo Bar');
+      expect(result).not.to.equal('Foo Bar');
```
Comment (peternhale) [inline packages/kit/test/nodash/internal.test.ts:104]: @maggiben Why is this test `not.equal` This seems to cover the case I mentioned above, "what happens if the string is already formatted in title case?"

But I would think that calling camelCaseToTitleCase with "Foo Bar" would return "Foo Bar".

Maybe test if the input string is properly formatted before transformation. If true then just return the input string.

**Review by peternhale — CHANGES_REQUESTED** [review]

File: packages/kit/test/nodash/internal.test.ts (hunk 3 above)
Comment (maggiben) [reply packages/kit/test/nodash/internal.test.ts:104]: @peternhale 

But I would think that calling camelCaseToTitleCase with "Foo Bar" would return "Foo Bar".

In reality it returns "Foo  Bar" with one extra space added in between the two words:

image

I copy the source from here: https://github.com/salesforcecli/plugin-org/blob/6f0d1c4093056b09e2c9fdf5501f83cdd8e9fa1e/src/shared/utils.ts#L17 didn't want to change it but if you need it to be so it strips the extra spaces I can do that.

Comment (peternhale) [reply packages/kit/test/nodash/internal.test.ts:104]: @maggiben Please make the change to ensure that resulting string has only one space between words.

Comment (maggiben) [reply packages/kit/test/nodash/internal.test.ts]: @peternhale done

File: packages/kit/src/nodash/internal.ts [hunk 4]
```diff
@@ -38,6 +38,19 @@ export function lowerFirst(value?: string): Optional<string> {
...
+/**
+ * Formats a camel case style `string` into a title case.
+ *
+ * @param text Text to transform.
+ */
+export function camelCaseToTitleCase(text: string): string {
+  return text
+    .replace(/(^\w|\s\w)/g, (m) => m.toUpperCase())
+    .replace(/([A-Z][a-z]+)/g, ' $1')
+    .replace(/  +/g, ' ')
```
Comment (peternhale) [inline packages/kit/src/nodash/internal.ts:50]: Just a nit
```suggestion
    .replace(/\s{2,}/g, ' ')
```

**Review by peternhale — APPROVED** [review]

**Review by peternhale — APPROVED** [review]

**Pull request merged**